### ISSUE TITLE
CA1822 / CA1024 False-positive for awaitable pattern

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#4623]
### Analyzer

- **Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822)
- **Diagnostic ID**: [CA1024](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1024)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.1 (Latest)

### Steps To Reproduce

````charp
using System;
using System.Runtime.CompilerServices;
using System.Threading.Tasks;

namespace Dummy
{
    public class Repro
    {
        public static async Task ReproAsync()
        {
            await new DummyAwaitable();
        }
    }

    public readonly struct DummyAwaitable : INotifyCompletion
    {
        // CA1024 Use properties where appropriate
        public DummyAwaitable GetAwaiter()
        {
            return this;
        }

        // CA1822 Member 'GetResult' does not access instance data and can be marked as static
        public void GetResult()
        {
        }

        // CA1822 Member 'IsCompleted' does not access instance data and can be marked as static
        public bool IsCompleted => false;

        public void OnCompleted(Action continuation) => throw null;
    }
}
````

### Expected behavior

No diagnostic

### Actual behavior

It reports diagnostics. If you apply the fix, the code doesn't compile.

**Comment by Evangelink** [comment]
Thanks for reporting these cases! 

I really hate magic methods... It just make everything insanely complex to handle.
